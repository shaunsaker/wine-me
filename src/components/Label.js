import React from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import styleConstants from "../assets/styleConstants";

import { Label } from "react-native-simple-components";

export default function LabelComponent(props) {
    /*
    static get propTypes() {
        return {
            text: PropTypes.string,
            highlight: PropTypes.bool,
            secondaryText: PropTypes.bool,
            handlePress: PropTypes.bool,
        };
    }
*/

    return (
        <Label
            text={props.text}
            textStyle={[
                styles.labelText,
                props.secondaryText && styles.secondaryLabelText,
            ]}
            style={[styles.label, props.highlight && styles.highlightedLabel]}
            handlePress={props.handlePress}
        />
    );
}

const styles = StyleSheet.create({
    label: {
        backgroundColor: styleConstants.transBlack,
        borderRadius: 4,
        alignSelf: "flex-start",
        paddingHorizontal: 16,
        marginRight: 4,
        marginBottom: 4,
    },
    highlightedLabel: {
        backgroundColor: styleConstants.primary,
    },
    labelText: {
        fontSize: styleConstants.smallFont,
        color: styleConstants.white,
        ...styleConstants.primaryFont,
    },
    secondaryLabelText: {
        fontSize: styleConstants.regularFont,
        ...styleConstants.secondaryFont,
    },
});
