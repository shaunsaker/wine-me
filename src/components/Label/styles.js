import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: styleConstants.colors.white,
    paddingVertical: styleConstants.dimensions.spacing.vertical / 2,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal / 2,
    ...styleConstants.shadows.small,
    borderRadius: styleConstants.dimensions.borderRadius,
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.primaryText,
    marginRight: styleConstants.dimensions.spacing.horizontal / 2,
  },
  text: {
    ...styleConstants.fonts.types.extraSmall,
  },
});

export default styles;
