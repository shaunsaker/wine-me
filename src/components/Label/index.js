import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';

import styles from './styles';

import Touchable from '../Touchable';

const propTypes = {
  handlePress: PropTypes.func,
  text: PropTypes.string,
  testID: PropTypes.string,
};

const defaultProps = {};

const Label = ({ handlePress, text, testID }) => {
  return (
    <Touchable
      onPress={handlePress}
      disabled={!handlePress}
      style={styles.container}
      testID={testID}
    >
      <Text style={styles.text}>{text}</Text>
    </Touchable>
  );
};

Label.propTypes = propTypes;
Label.defaultProps = defaultProps;

export default Label;
