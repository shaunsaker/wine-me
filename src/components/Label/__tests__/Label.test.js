import React from 'react';
import renderer from 'react-test-renderer';

import Label from '..';

describe('Label', () => {
  const handlePress = jest.fn();
  const text = 'Test';

  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(<Label handlePress={handlePress} text={text} />);

      expect(component).toMatchSnapshot();
    });
  });
});
