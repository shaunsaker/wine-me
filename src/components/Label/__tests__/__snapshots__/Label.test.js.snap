// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders renders 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#E23E57",
      "borderColor": null,
      "borderRadius": 5,
      "borderWidth": 0,
      "elevation": 2,
      "opacity": 1,
      "paddingHorizontal": 7.5,
      "paddingVertical": 5,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.17,
      "shadowRadius": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 11,
      }
    }
  >
    Test
  </Text>
</View>
`;
